agent/taskresource/ssmsecret/ssmsecret_test.go

```diff
@@ -0,0 +1,377 @@
+// +build unit
```

Comment:
Why you want me to add test cases for situation (1)? I have the test case that secrets in multiple regions are retrieved in multiple goroutines. In terms of reading, it uses more than one API call to retrieve values which is the same as (1). And in terms of writing, since we are locking on resource itself, it is still the same case as (1).

Reply:
this blob is left untested if you don't have situation (1): https://github.com/aws/amazon-ecs-agent/blob/8096c6791257f3f096c870e90ae7f38005219bb9/agent/taskresource/ssmsecret/ssmsecret.go#L306-L312

Reply:
Test cases added